code/parse/sexp.cpp

```diff
@@ -10593,7 +10597,7 @@ int eval_for_ship_collection(int arg_handler_node, int condition_node, int op_co
 					if (just_count)
 						num_valid_arguments++;
 					else
-						argument_vector.emplace_back(vm_strdup(shipp->ship_name), -1);
+						argument_vector.emplace_back(shipp->ship_name, -1);
```

Comment:
vm_strdup is removed here. What effect does that have, exactly?

Reply:
Duping the string is not needed because `ship_name` is a permanent buffer.  The argument to `test_argument_vector_for_condition` is changed to `DUP_NOT_NEEDED` accordingly.